src/BinSkim.Rules/RuleIds.cs

```diff
@@ -69,5 +69,8 @@ internal static class RuleIds
         // MachO Checks
         public const string EnablePositionIndependentExecutableMachO = "BA5001";
         public const string DoNotAllowExecutableStack = "BA5002";
+
+        // SourceLink Checks
```

Comment:
>SourceLink [](http://example.com/codeflow?start=11&length=10)

Ok, so the rule id is always tricky. Mostly our rule IDs are organized around program + debug data representation, BA2XXX are for Windows and BA3XXX are for *nix. Also, these domains currently are focused entirely on security.

Is SourceLink applicable only to MS-generated binaries? Is this data relevant for *nix platforms, say for .NET core? Will *nix or others build their own similar features for native *nix formats + package managers?

This feature does, in fact, have a security element to it: the inability to acquire source code for a compromised (often old) component is a real problem. It's why standards such as the SDL recommend archiving PDBs: persisting/accessing relevant source is an extension of this value.

So we could simply extend the current rule set and make this BA2027.

On the other hand, we could consider this rule the start of a new class of BinSkim analysis, which is more oriented towards general goodness, enabling debugging, performance optimizations, etc.

Reply:
BA2027 it is